processing/src/main/java/org/apache/druid/segment/DoubleDimensionIndexer.java

```diff
@@ -132,33 +132,32 @@ public DimensionSelector makeDimensionSelector(
       @Override
       public boolean isNull()
       {
-        final Object[] dims = currEntry.get().getDims();
-        return hasNulls && (dimIndex >= dims.length || dims[dimIndex] == null);
+        return hasNulls && currEntry.get().isDimNull(dimIndex);
```

Comment:
This change and the below changes are mainly for performance reasons, right?
to avoid the need to return all dims when they are off heap